### PR TITLE
Allow direct pot file creation

### PR DESCRIPTION
**Pull request by pgaultier**
add support for message 'format' => 'pot'

This allow the user to use tools like poedit to manage the translations

### CODE REVIEW

File: framework/console/controllers/MessageController.php [hunk 1]
```diff
@@ -133,21 +133,26 @@ public function actionExtract($configFile)
                     $this->saveMessagesToPHP($messages, $dir, $config['overwrite'], $config['removeUnused'], $config['sort']);
                 }
             }
-        } elseif ($config['format'] === 'db') {
-            $db = \Yii::$app->get(isset($config['db']) ? $config['db'] : 'db');
-            if (!$db instanceof \yii\db\Connection) {
-                throw new Exception('The "db" option must refer to a valid database application component.');
+        } elseif (in_array($config['format'], ['db', 'pot'])) {
+            if ($config['format'] === 'db') {
```
Comment (klimov-paul) [inline framework/console/controllers/MessageController.php]: `If` statements are inconsistent: why unite 'db' and 'pot' at first `if`, while they are completely split at second?

Comment (pgaultier) [inline framework/console/controllers/MessageController.php]: Ok, changed.
Originaly I kept them together as "pot" and "db" do not need to iterate over the languages

File: framework/console/controllers/MessageController.php [hunk 2]
```diff
@@ -586,4 +589,41 @@ protected function saveMessagesToPO($messages, $dirName, $overwrite, $removeUnus
             $this->stdout("Nothing to save.\n", Console::FG_GREEN);
         }
     }
+
+    /**
+     * Writes messages into POT file
+     *
+     * @param array $messages
+     * @param string $dirName name of the directory to write to
+     * @param string $catalog message catalog
```
Comment (samdark) [inline framework/console/controllers/MessageController.php]: Need `@since 2.0.5` here.

File: framework/console/controllers/MessageController.php [hunk 3]
```diff
@@ -586,4 +589,41 @@ protected function saveMessagesToPO($messages, $dirName, $overwrite, $removeUnus
...
+     */
+    protected function saveMessagesToPOT($messages, $dirName, $catalog)
+    {
+        $file = str_replace("\\", '/', "$dirName/$catalog.pot");
```
Comment (samdark) [inline framework/console/controllers/MessageController.php]: Is this replacement actually required?

Comment (pgaultier) [inline framework/console/controllers/MessageController.php]: Well replacement exists for PO messages. This is why I kept it.

Comment (samdark) [inline framework/console/controllers/MessageController.php]: OK.

**Comment by samdark** [comment]
Overall looks good to me.

**Comment by klimov-paul** [comment]
Fine with me.

**Comment by samdark** [comment]
Merged. Thanks!

**Comment by samdark** [comment]
@pgaultier email used in your commit is different from your github's one. If you want stats and contributors list to work correctly you can add additional emails in settings.

**Pull request closed without merging**